### PR TITLE
[language][test-generation] remove diem dependencies

### PR DESCRIPTION
**Pull request by vgao1996**
This makes test-generation no longer depend on diem crates. Specifically, DiemVM is replaced with MoveVM and diem-logger is replaced with tracing.

Note: I cannot really test this change since this tool has been broken since long time ago. Let's try to revive it in the future and hook it up to some testing backends.

### CODE REVIEW

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  M tracing 0.1.26 (transitive third-party, crates.io)
    * version upgraded from 0.1.25
    * (unchanged features: std)
  M tracing-core 0.1.18 (transitive third-party, crates.io)
    * version upgraded from 0.1.17
    * (unchanged features: lazy_static, std)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M tracing 0.1.26 (transitive third-party, crates.io)
    * version upgraded from 0.1.25
    * (unchanged features: log, std)
  M tracing-core 0.1.18 (transitive third-party, crates.io)
    * version upgraded from 0.1.17
    * (unchanged features: lazy_static, std)
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  M tracing 0.1.26 (transitive third-party, crates.io)
    * version upgraded from 0.1.25
    * (unchanged features: std)
  M tracing-core 0.1.18 (transitive third-party, crates.io)
    * version upgraded from 0.1.17
    * (unchanged features: lazy_static, std)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M tracing 0.1.26 (transitive third-party, crates.io)
    * version upgraded from 0.1.25
    * (unchanged features: log, std)
  M tracing-core 0.1.18 (transitive third-party, crates.io)
    * version upgraded from 0.1.17
    * (unchanged features: lazy_static, std)
```

**Review by tzakian — DISMISSED** [review]
LGTM

**Comment by vgao1996** [comment]
@tzakian could you restamp this? Had to rebase. Thanks!

File: language/testing-infra/test-generation/src/lib.rs [hunk 1]
```diff
@@ -49,6 +55,23 @@ fn run_verifier(module: CompiledModule) -> Result<CompiledModule, String> {
...
 }
 
+static STORAGE_WITH_MOVE_STDLIB: Lazy<InMemoryStorage> = Lazy::new(|| {
+    let mut storage = InMemoryStorage::new();
+    let (_, compiled_units) = Compiler::new(&move_stdlib::move_stdlib_files(), &[])
+        .build_and_report()
+        .unwrap();
+    let compiled_modules = compiled_units.into_iter().map(|unit| match unit {
+        CompiledUnit::Module { module, .. } => module,
+        CompiledUnit::Script { .. } => panic!("Unexpected Script in stdlib"),
```
Comment (modocache) [inline language/testing-infra/test-generation/src/lib.rs:65]: 👍

**Review by modocache — DISMISSED** [review]
I took a look myself, and hopefully this can work in place of a re-submitted review from @tzakian.

**Comment by vgao1996** [comment]
Thanks @modocache 

/land

**Pull request closed without merging**